File: T11 Programación orientada a objetos/Actividades/calculadora.py
class Calculadora:
    def __init__(self):
        self.num1 = 0
        self.num2 = 0
        self.opcion = 0

    def pedir_datos(self):
        self.opcion = int(input("\n¿Qué operación deseas realizar? \
            \n 1. Suma \n 2. Resta \n 3. Multiplicación \n 4. División \n 5. Salir \n "))
        if self.opcion in range(1, 5):
            self.num1 = int(input("Introduce el primer número: "))
            self.num2 = int(input("Introduce el segundo número: "))
    def calcular(self):
        while True:
            if self.opcion == 1:
                resultado = self.num1 + self.num2
                print("\n El resultado de la suma es:", resultado)
            elif self.opcion == 2:
                resultado = self.num1 - self.num2
                print("\n El resultado de la resta es:", resultado)
            elif self.opcion == 3:
                resultado = self.num1 * self.num2
                print("\n El resultado de la multiplicación es:", resultado)
            elif self.opcion == 4:
                if self.num1 == 0 or self.num2 == 0:
                    print("\n No se puede dividir por 0")
                else:
                    resultado = self.num1 / self.num2
                    print("\n El resultado de la división es:", resultado)
            elif self.opcion == 5:
                print("\n ¡Hasta Luego!")
                break
            else:
                print("\n Opción no válida")
            self.pedir_datos()

calculadora = Calculadora()
calculadora.pedir_datos()
calculadora.calcular()

File: T13 Introduccion a JAVA/Calculadora.java
import java.util.Scanner;

public class Calculadora {
    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            String opcion;

            while (true) {
                System.out.println("\nElija la operación a realizar:");
                System.out.println("1) Suma");
                System.out.println("2) Resta");
                System.out.println("3) Multiplicación");
                System.out.println("4) División");
                System.out.println("5) División entera");
                System.out.println("6) Módulo");
                System.out.println("7) Potencia");
                System.out.println("8) Puerta AND a nivel de bit");
                System.out.println("9) Puerta OR a nivel de bit");
                System.out.println("10) Salir");

                opcion = scanner.next();

                if(opcion.equals("10")) {
                    System.out.println("¡Hasta luego!");
                    return;
                }

                System.out.print("Ingrese el primer número: ");
                double num1 = scanner.nextDouble();
                System.out.print("Ingrese el segundo número: ");
                double num2 = scanner.nextDouble();

                double resultado = 0;

                switch(opcion) {
                    case "1":
                        resultado = num1 + num2;
                        break;
                    case "2":
                        resultado = num1 - num2;
                        break;
                    case "3":
                        resultado = num1 * num2;
                        break;
                    case "4":
                        if(num1 == 0 || num2 == 0) {
                            System.out.println("Error: no se puede dividir entre cero.");
                        } else {
                            resultado = num1 / num2;
                        }
                        break;
                    case "5":
                        if(num1 == 0 || num2 == 0) {
                            System.out.println("Error: no se puede dividir entre cero.");
                        } else {
                            resultado = (int) (num1 / num2);
                        }
                        break;
                    case "6":
                        resultado = num1 % num2;
                        break;
                    case "7":
                        resultado = Math.pow(num1, num2);
                        break;
                    case "8":
                        resultado = (byte) num1 & (byte) num2;
                        break;
                    case "9":
                        resultado = (byte) num1 | (byte) num2;
                        break;
                    default:
                        System.out.println("Opción inválida.");
                        continue; // repetir el ciclo
                }

                System.out.println("El resultado es: " + resultado);
            }
        }
    }
}
